### PR TITLE
bigquery: complete system tests

### LINKED ISSUE #2000: bigquery: complete system tests
Here's what we need to hit.

References:
- [C# tests](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/ExhaustiveTypesTest.cs)
- [`schema.json` and test data](https://gist.github.com/stephenplusplus/d9a0b1733a57381026b6222df5a60639)

| Data type                                | Queries (Legacy) | Queries (Standard) | tabledata.list | tabledata.insertAll | Creating a table | Query Params (Beta) |
|------------------------------------------|------------------|--------------------|----------------|---------------------|------------------|---------------------|
| scalar value columns                     |                  |                    |                |                     |                  |                     |
| repeated scalar value columns            | N/A (flattened)  |                    |                |                     |                  |                     |
| record value columns                     | N/A (flattened)  |                    |                |                     |                  |                     |
| repeated record value columns            | N/A (flattened)  |                    |                |                     |                  |                     |
| record value inside another record value | N/A (flattened)  |                    |                |                     |                  |                     |
| repeated value within a record value     | N/A (flattened)  |                    |                |                     |                  |                     |
| timestamp (UTC)                          |                  |                    |                |                     |                  |                     |
| datetime (no time zone)                  | N/A?             |                    |                |                     |                  |                     |
| date                                     | N/A?             |                    |                |                     |                  |                     |
| time                                     | N/A?             |                    |                |                     |                  |                     |
| bytes                                    | N/A?             |                    |                |                     |                  |                     |

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #2000 

cc @tswast

### CODE REVIEW

**Comment by callmehiphop** [comment]
Should we expect a review from some one on the BigQuery team?

**Comment by tswast** [comment]
I wouldn't count on it, but I can try to recruit someone if you'd like. For what it's worth, the expected rows in `packages/bigquery/system-test/bigquery.js` were precisely what I was after in suggesting the tests. It makes me a lot more comfortable to see these data types are coming through properly.

**Comment by callmehiphop** [comment]
@tswast cool, thanks for the heads up!

File: packages/bigquery/src/table.js [hunk 1]
```diff
@@ -258,7 +258,7 @@ Table.encodeValue_ = function(value) {
  * @param {array} rows
  * @return {array} Fields using their matching names from the table's schema.
  */
-Table.mergeSchemaWithRows_ = function(schema, rows) {
+Table.mergeSchemaWithRows_ = function(BigQuery, schema, rows) {
```
Comment (callmehiphop) [inline packages/bigquery/src/table.js]: Why do we need to pass this in? Seems like something that we could easily require.

Comment (stephenplusplus) [inline packages/bigquery/src/table.js]: Cross-dependency blow-up :(

**Comment by callmehiphop** [comment]
@stephenplusplus might be unrelated to the PR, but locally I'm getting an error where the system tests start to pass, but when I get to streaming inserts I get this

```sh
  1) BigQuery BigQuery/Table should insert rows via stream:
     Error: Sorry, we cannot connect to Google Cloud Services without a project ID. You may specify one with an environment variable named "GCLOUD_PROJECT". See https://googlecloudplatform.github.io/google-cloud-node/#/docs/guides/authentication for a detailed guide on creating an authenticated connection.
      at Object.<anonymous> (node_modules/@google-cloud/common/src/util.js:55:29)
      at Object.<anonymous> (node_modules/@google-cloud/common/src/service-object.js:32:12)
      at Object.<anonymous> (node_modules/@google-cloud/common/src/operation.js:31:21)
      at Object.<anonymous> (node_modules/@google-cloud/common/src/index.js:27:21)
      at Object.<anonymous> (src/dataset.js:23:14)
      at Object.<anonymous> (system-test/bigquery.js:25:15)
      at Array.forEach (native)
      at node.js:814:3
```

**Comment by stephenplusplus** [comment]
That is strange. Can you open a new issue for that?

**Pull request merged**